### ISSUE TITLE
Unique constraint to ignore null values

### ISSUE THREAD

**Issue by bantolov** [Rhetos/Rhetos#177]
**Is your feature request related to a problem? Please describe.**
The `Unique` concept does not allow entering more than one record with NULL value.
Sometimes we need to allow multiple null values, and only not-null values should be unique.

This feature is based on question #130.

**Describe the solution you'd like**
Create a new concept: `WhereNotNull` as a part of standard CommonConcepts library.
WhereNotNull will be used as an addition on the Unique concept, to limit the unique constraint to check only the values that are not null.

Unique concept just creates a UNIQUE INDEX on the database column. **The UNIQUE INDEX in SQL Server does not allow two NULL values in the column.**
WhereNotNull concept should simply add `WHERE ColumnName IS NOT NULL` to the index created by Unique concept.

Usage example:

```c
Entity Partner
{
    ShortString VAT { Unique { WhereNotNull; } }
}
```

**Describe alternatives you've considered**
Some workarounds are described in question #130.

**Comment by bantolov** [comment]
WhereNotNull concept (as proposed above) should be made more flexible, since the database allows custom filters for unique constraints. With same effort, we can allow any SQL filter that is supported by the unique constraint in db, with a `Where` concept from the following example.

```c
Entity Partner
{
    ShortString VAT { Unique { Where 'VAT IS NOT NULL'; } }
}
```

On the other hand, to make the code more readable in the most common scenario (unique for not null), we can create a more specific concept `UniqueIfNotNull`, for example:

```c
Entity Partner
{
    ShortString VAT { UniqueIfNotNull; }
}
```

**Comment by bantolov** [comment]
Closed by #476
